packages/callout-block/src/CalloutBlock.tsx

```diff
@@ -88,11 +91,11 @@ export const CalloutBlock: FC<CalloutBlockProps> = ({ appBridge }) => {
                     ...customCornerRadiusStyle,
                 }}
             >
-                {blockSettings.iconSwitch && iconUrl && (
-                    <span className="tw-mr-3 tw-flex-none tw-w-6 tw-h-6">
-                        <img data-test-id="callout-icon" alt="Callout icon" src={iconUrl} />
-                    </span>
-                )}
+                <CalloutIcon
+                    iconType={blockSettings.iconSwitch ? Icon.Custom : blockSettings.iconType}
+                    iconUrl={iconUrl}
+                    hasValue={hasRichTextValue(blockSettings.textValue)}
+                />
```

Comment:
I would keep the condition to show the icon outside of the `CalloutIcon` component. If there is no icon, we also don't need to run `hasRichTextValue`.